src/stan/services/init/initialize_state.hpp

```diff
@@ -113,52 +114,44 @@ namespace stan {
        *                            right size and set to 0.
        * @param[in,out] model       the model. Side effects on model? I'm not
        *                            quite sure
-       * @param[in,out] output      output stream for messages
+       * @param[in,out] writer      writer callback for messages
```

Comment:
I guess given how little this function will be called the template isn't necessary and we can go with virtual functions.